File: sample_daggerandroid/src/main/java/io/github/hurshi/sample_daggerandroid/MainActivity.java
package io.github.hurshi.sample_daggerandroid;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentTransaction;
import android.util.Log;

import javax.inject.Inject;

import dagger.android.support.DaggerAppCompatActivity;
import io.github.hurshi.sample_daggerandroid.beans.ActivityBean;
import io.github.hurshi.sample_daggerandroid.beans.AppBean;
import io.github.hurshi.sample_daggerandroid.modules.ActivityModule;
import io.github.hurshi.sample_daggerandroid.scopes.ActivityScope;
import io.github.hurshi.sample_daggerandroid.scopes.FragmentScope;
import io.github.hurshi.simplifydagger.annotation.AutoAndroidComponent;


@AutoAndroidComponent(scope = ActivityScope.class, modules = {ActivityModule.class}, subScope = FragmentScope.class)
public class MainActivity extends DaggerAppCompatActivity {
    @Inject
    ActivityBean activityBean;

    @Inject
    AppBean appBean;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.e(">>>", "log from MainActivity  " + appBean.toString());
        Log.e(">>>", "log from MainActivity  " + activityBean.toString());

        loadFragment();
    }

    private void loadFragment() {
        TestFragment fragment = new TestFragment();
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.add(R.id.frame_layout, fragment);
        transaction.commit();
    }
}

File: sample_architecture_components/src/main/java/io/github/hurshi/sample_architecture_components/MainActivity.kt
package io.github.hurshi.sample_architecture_components

import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.util.Log
import dagger.android.support.DaggerAppCompatActivity
import io.github.hurshi.sample_architecture_components.entity.DaoSessionSimulate
import io.github.hurshi.sample_architecture_components.scopes.ActivityScope
import io.github.hurshi.sample_architecture_components.scopes.AppScope
import io.github.hurshi.sample_architecture_components.vm.MainActivityViewModel
import io.github.hurshi.sample_architecture_components.vm_factory.VMFactory
import io.github.hurshi.simplifydagger.annotation.AutoAndroidComponent
import io.github.hurshi.simplifydagger.annotation.AutoViewModelFactoryComponent
import javax.inject.Inject

@AutoViewModelFactoryComponent(factory = VMFactory::class, viewModel = MainActivityViewModel::class, factoryScope = AppScope::class)
@AutoAndroidComponent(scope = ActivityScope::class)
class MainActivity : DaggerAppCompatActivity() {

    @Inject
    lateinit var factory: ViewModelProvider.Factory

    @Inject
    lateinit var daoSessionSimulate: DaoSessionSimulate

    private val vm: MainActivityViewModel by lazy {
        ViewModelProviders.of(this@MainActivity, factory).get(MainActivityViewModel::class.java)
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Log.e(">>>", "daoSessionSimulate = $daoSessionSimulate")
        Log.e(">>>", "daoSessionSimulate from vm = ${vm.getDaoSessionSimulate()}")
    }
}